api/all.yaml

```diff
@@ -1414,6 +1702,39 @@ components:
           description: |
             Opaque token to retrieve the next page of results. Absent if there are no more pages.
             __page_token__ should be set to this value for the next request (for the next page of results).
+    GenerateTemporaryModelCredential:
+      example:
+        model_id: model_id
+        operation: null
+      properties:
+        model_id:
+          description: |
+            Model id for which temporary credentials are generated. 
+            Can be obtained from models/{full_name} (get model info) API.
+          type: string
+        operation:
+          $ref: '#/components/schemas/ModelOperation'
+      type: object
+      required:
+        - model_id
+        - operation
+    GenerateTemporaryModelCredentialResponse:
+      properties:
+        aws_temp_credentials:
```

Comment:
Basic question, is Azure not in the picture here? Why only aws_temp_credentials?

Reply:
They are working on adding the other credentials.  In DB UC, this is basically a TemporaryCredentials message where the credentials field looks like:

```  oneof credentials {
    AwsCredentials aws_temp_credentials = 2;
    AzureUserDelegationSAS azure_user_delegation_sas = 3;
    GcpOauthToken gcp_oauth_token = 4;
    ForeignCredential foreign_credential = 5;
    AzureActiveDirectoryToken azure_aad = 6;
    R2Credentials r2_temp_credentials = 8;
  }```

Reply:
We should define a common message for credential in UC OSS, and share it cross all genreateTempCred APIs. cc. @alexreid-db

Reply:
@haogang I'd agree, but this would deviate from DB UC apis which for some reason defines separate requests/responses/components even though they technically don't differ in their definitions.

Reply:
Why? DB UC APIs has `TemporaryCredentials` message that are shared by all the response messages.